drivers/led_strip/ws2812_i2s.c

```diff
@@ -51,43 +49,32 @@ struct ws2812_i2s_cfg {
 /* Serialize an 8-bit color channel value into two 16-bit I2S values (or 1 32-bit
  * word).
  */
-static inline void ws2812_i2s_ser(uint32_t *word, uint8_t color, const uint8_t sym_one,
-				  const uint8_t sym_zero)
+static inline uint32_t ws2812_i2s_ser(uint8_t color, const uint8_t sym_one,
+				      const uint8_t sym_zero)
 {
-	*word = 0;
-	for (uint16_t i = 0; i < 8; i++) {
-		if ((1 << i) & color) {
-			*word |= sym_one << (i * 4);
-		} else {
-			*word |= sym_zero << (i * 4);
-		}
+	uint32_t word = 0;
+
+	for (uint_fast8_t mask = 0x80; mask != 0; mask >>= 1) {
+		word <<= 4;
+		word |= (color & mask) ? sym_one : sym_zero];
```

Comment:
ohhh. That was not the idea. Just a moment

Reply:
Hi @GardeningStevie. Thanks for this patch. Note that I expect you to build and test the code you submit. Typo happens, there is no problem with that. But since the same thing happens on another PR (#pullrequestreview-2464635105), I have to say it.

Reply:
Hi @simonguinot ,
Agree. I was very lazzy with these PRs. Sorry for that. Essentially some code lines were just copy & pasted from our project Zephyr repository branch.
What could go wrong? Yes, a lot. E.g. clang-format still complains about my `ws2812_i2s_ser` function.